Tests/SonarScanner.MSBuild.Tasks.IntegrationTests/TargetsTests/SonarResolveReferencesTests.cs

```diff
@@ -0,0 +1,79 @@
+﻿/*
+ * SonarScanner for MSBuild
+ * Copyright (C) 2016-2021 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.IO;
+using System.Linq;
+using FluentAssertions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using SonarScanner.MSBuild.Common;
+using SonarScanner.MSBuild.Tasks;
+using SonarScanner.MSBuild.Tasks.IntegrationTests;
+using TestUtilities;
+
+namespace SonarScanner.Integration.Tasks.IntegrationTests.TargetsTests
+{
+    [TestClass]
+    public class SonarResolveReferencesTests
+    {
+        public TestContext TestContext { get; set; }
+
+        [TestMethod]
+        [TestCategory("IsTest")]
+        public void BuildIntegration_ResolvesReferences()
+        {
+            // Arrange
+            var rootFolder = TestUtils.CreateTestSpecificFolderWithSubPaths(TestContext);
+            var projectSnippet = $@"
+<PropertyGroup>
+    <SonarQubeTempPath>{rootFolder}</SonarQubeTempPath>
+</PropertyGroup>";
+
+            var filePath = CreateProjectFile(projectSnippet);
+
+            // Act
+            var result = BuildRunner.BuildTargets(TestContext, filePath, TargetConstants.DefaultBuildTarget);
+
+            // Assert
+            result.AssertTargetSucceeded(TargetConstants.DefaultBuildTarget);
+            result.AssertTargetExecuted(TargetConstants.ResolveReferencesTarget);
+            result.AssertTargetExecuted(TargetConstants.CategoriseProjectTarget);
+
+            var sonarResolvedReferences = result.GetCapturedItemValues(TargetProperties.SonarResolvedReferences);
+            sonarResolvedReferences.Should().NotBeEmpty();
+            sonarResolvedReferences.Should().Contain(x => x.Value.Contains("mscorlib"));
+        }
+
+        private string CreateProjectFile(string projectSnippet)
+        {
+            var captureReferences = $@"
+<Project ToolsVersion='Current' xmlns='http://schemas.microsoft.com/developer/msbuild/2003'>
+  <Target Name='CaptureValues' AfterTargets='{TargetConstants.CategoriseProjectTarget}'>
+    <Message Importance='high' Text='CAPTURE::ITEM::{TargetProperties.SonarResolvedReferences}::%({TargetProperties.SonarResolvedReferences}.Identity)' />
```

Comment:
why is this `Identity` suffix needed?

Reply:
That's the MSBuild way how to access the value included in the ItemGroup itself, see [Docs](https://docs.microsoft.com/en-us/visualstudio/msbuild/msbuild-well-known-item-metadata?view=vs-2019).
It's inspired by this:
https://github.com/SonarSource/sonar-scanner-msbuild/blob/0779f4d58eec352021ce36e32fe8df574c78bba8/Tests/SonarScanner.MSBuild.Tasks.IntegrationTests/Resources/CaptureDataTargetsFileTemplate.xml#L18